### PR TITLE
Add the ability to query similarity index by event.

### PR DESCRIPTION
**Pull request by tkaemming**
#nochanges

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by tkaemming** [comment]
I'm going to rewrite some of the comments for the refactored Lua bits, but otherwise this is pretty much stable.

File: src/sentry/scripts/similarity/index.lua [hunk 1]
```diff
@@ -281,44 +281,215 @@ local function collect_index_key_pairs(arguments, validator)
...
+                        configuration,
+                        index,
+                        time,
+                        band,
+                        bucket
+                    )
+                )
+                for _, member in ipairs(members) do
+                    -- TODO: Count the number of bands that we've collided in
+                    -- to allow setting thresholds here.
```
Comment (tkaemming) [inline src/sentry/scripts/similarity/index.lua]: This should happen before we put this on the ingest path.

**Review by tkaemming — COMMENTED** [review]
I think this is in a good spot now.

**Review by JTCunning — APPROVED** [review]
:female_detective:

**Pull request merged**